src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -247,6 +247,7 @@ protected ImmutableArray<LocalSymbol> BuildLocals(SyntaxList<StatementSyntax> st
                     case SyntaxKind.YieldReturnStatement:
                     case SyntaxKind.ReturnStatement:
                     case SyntaxKind.ThrowStatement:
+                    case SyntaxKind.GotoCaseStatement:
```

Comment:
>SyntaxKind.GotoCaseStatement [](start = 25, length = 28)

Should we also handle regular GotoStatement?

Reply:
The parser cannot produce a node for a regular goto statement that has this issue.